packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -47,9 +49,27 @@ class TestCommand extends FlutterCommand {
       testArgs.insert(0, '--no-color');
     List<BuildConfiguration> configs = buildConfigurations;
     bool foundOne = false;
+
+    if (!FileSystemEntity.isFileSync(path.join(flutterDir.path, 'pubspec.yaml'))) {
+      print('${flutterDir.path} has no pubspec.yaml');
+      return 1;
+    }
+
+    if (!FileSystemEntity.isFileSync(path.join(flutterDir.path, 'pubspec.lock'))) {
+      print("Running pub get...");
+      Process process = await Process.start(
+        sdkBinaryName('pub'), ['get'],
+        workingDirectory: flutterDir.path
+      );
+      stdout.addStream(process.stdout);
+      stderr.addStream(process.stderr);
```

Comment:
I think process.dart has a wrapper that does this.  Something about "stream output"